src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapGroupBySettings.jsx

```diff
@@ -29,26 +32,160 @@ const HeatmapGroupBySettings = () => {
     );
   };
 
+  const getCellSetsOrder = () => {
+    const allCellSetsGroupBys = getCellSets(['cellSets', 'metadataCategorical']);
+
+    let groupedCellSets = [];
+
+    // from the enabled cell sets keys we get, find their corresponding information
+    groupedTracksKeys
+      .forEach((trackKey) => {
+        const groupBy = allCellSetsGroupBys.find((cellSetGroupBy) => cellSetGroupBy.key === trackKey);
+        groupedCellSets.push(groupBy);
+      });
+
+    // About the filtering: If we have failed to find some of the groupbys information, 
+    // then ignore those (this is useful for groupbys that sometimes dont show up, like 'samples')
+    return groupedCellSets.filter((groupedCellSet) => groupedCellSet !== undefined);
+  };
+
+  const isInitialRenderRef = useRef(true);
+  const [cellSetsOrder, setCellSetsOrder] = useState(getCellSetsOrder());
+
+  useEffect(() => {
+    if (isInitialRenderRef.current) {
+      isInitialRenderRef.current = false;
+
+      return;
+    }
+
+    if (cellSetsOrder.length === 0) {
+      return;
+    }
+
+    dispatch(
+      updatePlotConfig('interactiveHeatmap', {
+        groupedTracks: cellSetsOrder.map((cellSet) => cellSet.key)
+      })
+    );
+  }, [cellSetsOrder]);
+
+  const moveUp = (source, id) => {
+    const index = source.findIndex((e) => e.key === id);
+
+    const arr = [...source];
+
+    if (index <= 0) {
+      return arr;
+    }
+
+    const el = arr[index];
+    arr[index] = arr[index - 1];
+    arr[index - 1] = el;
+
+    return arr;
+  };
+
+  const moveDown = (source, id) => {
+    const index = source.findIndex((e) => e.key === id);
+
+    const arr = [...source];
+
+    if (index === -1 || index >= source.length - 1) {
+      return arr;
+    }
+
+    const el = arr[index];
+    arr[index] = arr[index + 1];
+    arr[index + 1] = el;
+
+    return arr;
+  };
```

Comment:
this and lots of other code is essentially duplicated code from the `HeatmapMetadataTrackSettings` component in order to create this component with the up and down arrows that is used in here and in `HeatmapMetadataTrackSettings`. 
I would suggest that you extract this in a new, separate component. If possible, make it generic, like `ColorPicker`, `FocusButton` for example that do not depend on specific state in Redux. In that way, we would be able to easily reuse it anywhere in the UI in the future. If it can be made generic, place it in the place where we keep such components, which is in `src/components`. Make sure that you use this new components in `HeatmapGroupBySettings` and in  `HeatmapMetadataTrackSettings` files.